dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/MetadataResourceHelper.java

```diff
@@ -34,10 +36,18 @@ public static String createWorkflowURL(WorkflowPath workflow, String entryType)
         return String.format("%s/%ss/%s", baseUrl, entryType, workflow.getEntryPath());
     }
 
+    public static String createWorkflowURL(RSSWorkflowPath workflow, String entryType) {
```

Comment:
Interesting, normally I would stick these two methods in RSSToolPath.java  and the workflow one, so they know how to generate their own paths.

Reply:
So they can generate the paths, they just don't know how to generate the URL

Reply:
Maybe they can both generate the rest of the path and whatever is calling createWorkflowURL can prepend the baseURL in both cases. 
This is a bit of a sideshow though, I see what you mean about the class proliferation